src/main/java/io/zipcoder/Student.java

```diff
@@ -1,4 +1,116 @@
 package io.zipcoder;
 
-public class Student {
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Student implements Comparable<Student> {
+
+    String firstName;
+    String lastName;
+    ArrayList<Double> examScores;
+
+    public Student(String aFirstName, String aLastName, Double[] testScores) {
+        this.firstName = aFirstName;
+        this.lastName = aLastName;
+        this.examScores = new ArrayList<>();
+        this.examScores.addAll(Arrays.asList(testScores));
+    }
+
+    public String getFirstName() {
+
+        return firstName;
+    }
+
+    public void setFirstName(String newFirstName) {
+
+        this.firstName = newFirstName;
+    }
+
+    public String getLastName() {
+
+        return lastName;
+    }
+
+    public void setLastName(String newLastName) {
+
+        this.lastName = newLastName;
+    }
+
+
+    public int getNumberOfExamsTaken() {
+
+        return examScores.size();
+    }
+
+    public String getExamScores() {
+        StringBuilder toAppendExamScores = new StringBuilder();
+        for (int i = 0; i < examScores.size(); i++) {
+            toAppendExamScores.append("\tExam " + (i + 1) + " -> " + examScores.get(i) + "\n");
+        }
+
+        return toAppendExamScores.toString();
+    }
+
+    public void addExamScore(double examScore) {
+        examScores.add(examScore);
+
+    }
+
+    public void setExamScore(int examNumber, double newScore) {
+        examScores.set(examNumber - 1, newScore);
+    }
+
+    public double getAverageExamScore() {
+
+        double totalScore = 0;
+        for (Double indexScore : this.examScores) {
+            totalScore += indexScore;
+        }
+
+        double roundAndMoveDecimal = Math.round(totalScore / examScores.size() * 100);
+
+        double averageExamScore = roundAndMoveDecimal / 100;
+
```

Comment:
you could do this cleaner.

Reply:
As before, I had tried to do this with one expression but it was not returning the result I needed.